packages/aws-cdk/lib/notices.ts

```diff
@@ -0,0 +1,270 @@
+          resolve([]);
+        }
+      });
+    });
+  }
+}
+
+interface CachedNotices {
+  expiration: number,
+  notices: Notice[],
+}
+
+const TIME_TO_LIVE = 60 * 60 * 1000; // 1 hour
+
+export class CachedDataSource implements NoticeDataSource {
+  constructor(
+    private readonly fileName: string,
+    private readonly dataSource: NoticeDataSource,
+    private readonly skipCache?: boolean) {
+  }
+
+  async fetch(): Promise<Notice[]> {
+    const cachedData = await this.load();
+    const notices = cachedData.notices;
+    const expiration = cachedData.expiration ?? 0;
+
+    if (Date.now() > expiration || this.skipCache) {
+      const freshData = {
+        expiration: Date.now() + TIME_TO_LIVE,
+        notices: await this.dataSource.fetch(),
+      };
+      await this.save(freshData);
+      return freshData.notices;
+    } else {
+      return notices;
+    }
+  }
+
+  private async load(): Promise<CachedNotices> {
+    try {
+      return await fs.readJSON(this.fileName) as CachedNotices;
+    } catch (e) {
+      debug(`Failed to load notices from cache: ${e}`);
+      return {
+        expiration: 0,
+        notices: [],
+      };
+    }
+  }
+
+  private async save(cached: CachedNotices): Promise<void> {
+    try {
+      await fs.writeJSON(this.fileName, cached);
+    } catch (e) {
+      debug(`Failed to store notices in the cache: ${e}`);
+    }
+  }
+}
+
+export interface NoticeFilterProps {
+  permanentlySuppressed: boolean,
+  temporarilySuppressed: boolean,
+  cliVersion: string,
+  frameworkVersion: string | undefined,
+  acknowledgedIssueNumbers: Set<number>,
+}
+
+export class NoticeFilter {
+  private readonly acknowledgedIssueNumbers: Set<number>;
+
+  constructor(private readonly props: NoticeFilterProps) {
+    this.acknowledgedIssueNumbers = props.acknowledgedIssueNumbers;
+  }
+
+  /**
+   * Returns true iff we should show this notice.
+   */
+  apply(notice: Notice): boolean {
+    if (this.props.permanentlySuppressed
+      || this.props.temporarilySuppressed
+      || this.acknowledgedIssueNumbers.has(notice.issueNumber)) {
+      return false;
+    }
+    return this.applyVersion(notice, 'cli', this.props.cliVersion) ||
+      this.applyVersion(notice, 'framework', this.props.frameworkVersion);
+  }
+
+  /**
+   * Returns true iff we should show the notice.
+   */
+  private applyVersion(notice: Notice, name: string, compareToVersion: string | undefined) {
+    if (compareToVersion === undefined) { return false; }
+
+    const affectedComponent = notice.components.find(component => component.name === name);
+    const affectedRange = affectedComponent?.version;
+    return affectedRange != null && semver.satisfies(compareToVersion, affectedRange);
+  }
+}
+
+function formatNotice(notice: Notice): string {
+  const componentsValue = notice.components.map(c => `${c.name}: ${c.version}`).join(', ');
+  return [
+    `${notice.issueNumber}\t${notice.title}`,
```

Comment:
I think the notice formatting in the RFC was a little more elaborate and optimized for readability :).